Sources/EmojiText/Render/AttributedStringEmojiRenderer.swift

```diff
@@ -0,0 +1,128 @@
+    }
+
+    func renderAnimated(emojis: [String: LoadedEmoji], size: CGFloat?, at time: CFTimeInterval) -> SwiftUI.Text {
+        let attributedString = renderAttributedString(with: emojis)
+
+        var result = Text(verbatim: "")
+        var partialString = AttributedPartialstring()
+
+        for run in attributedString.runs {
+            print(run)
+            if let emoji = run.attributes[EmojiAttribute.self] {
+                // If the run is an emoji we render it as an interpolated image in a Text view
+                let text = Text(emoji, size: size, at: time)
+
+                // If the same emoji is added multiple times in a row the run gets merged into one
+                // with their shortcodes joined. Therefore we simply divide distance of the range by
+                // the character count of the emojo to calculate how often the emoji needs to be displayed
+                let distance = attributedString.distance(from: run.range.lowerBound, to: run.range.upperBound)
+                let count = emoji.shortcode.count + 2
+
+                print(distance, count)
+                if distance == count {
+                    // Emoji is only displayed once
+                    result = [
+                        result,
+                        Text(&partialString),
+                        text
+                    ]
+                        .compactMap { $0 }
+                        .joined()
+                } else {
+                    // Emojis is displayed multiple times
+                    result = [
+                        result,
+                        Text(&partialString),
+                        Text(repating: text, count: distance / count)
+                    ]
+                        .compactMap { $0 }
+                        .joined()
+                }
+            } else {
+                // Otherwise we just append the run to AttributedPartialstring
+                partialString.append(attributedString[run.range])
+            }
+        }
+
+        return [result, Text(&partialString)]
+            .compactMap { $0 }
+            .joined()
+    }
+
+    private func renderAttributedString(with emojis: [String: LoadedEmoji]) -> AttributedString {
+        var string = attributedString
+        for (shortcode, emoji) in emojis {
+            if let range = attributedString.range(of: ":\(shortcode):") {
+                string[range].emoji = emoji
+            }
+        }
+        return string
+    }
+
+    func render(emojis: [String : LoadedEmoji], size: CGFloat?) -> NSAttributedString {
+        return NSAttributedString(attributedString)
+    }
+}
+
+extension AttributeScopes {
+    struct EmojiAttributes: AttributeScope {
+        let emoji: EmojiAttribute
+    }
+
+    var emoji: EmojiAttributes.Type { EmojiAttributes.self }
+}
+
+enum EmojiAttribute: AttributedStringKey {
+    typealias Value = LoadedEmoji
+    static let name = "Emoji"
+}
+
+extension AttributeDynamicLookup {
```

Comment:
Consider making these extensions private, since they seem specifically for `AttributedStringEmojiRenderer`

```suggestion
private extension AttributeScopes {
    struct EmojiAttributes: AttributeScope {
        let emoji: EmojiAttribute
    }

    var emoji: EmojiAttributes.Type { EmojiAttributes.self }
}

private enum EmojiAttribute: AttributedStringKey {
    typealias Value = LoadedEmoji
    static let name = "Emoji"
}

private extension AttributeDynamicLookup {
```